src/sentry/incidents/endpoints/organization_alert_rule_available_action_index.py

```diff
@@ -11,20 +11,41 @@
 from sentry.incidents.endpoints.serializers import action_target_type_to_string
 from sentry.incidents.logic import get_available_action_integrations_for_org
 from sentry.incidents.models import AlertRuleTriggerAction
+from sentry.models import OrganizationIntegration, PagerDutyService
 
 
 class OrganizationAlertRuleAvailableActionIndexEndpoint(OrganizationEndpoint):
-    def build_action_response(self, registered_type, integration=None):
-        return {
+    def fetch_pagerduty_services(self, organization, integration_id):
+        services = PagerDutyService.objects.filter(
+            organization_integration=OrganizationIntegration.objects.get(
+                organization=organization, integration=integration_id
+            )
+        )
+        formatted_services = [
+            {"value": service.id, "label": service.service_name} for service in services
+        ]
+        return formatted_services
+
+    def build_action_response(self, organization, registered_type, integration=None):
+        allowed_target_types = [
+            action_target_type_to_string[target_type]
+            for target_type in registered_type.supported_target_types
+        ]
+
+        input_type = "select" if registered_type.slug in ["pagerduty", "email"] else "text"
+
+        action_response = {
             "type": registered_type.slug,
-            "allowedTargetTypes": [
-                action_target_type_to_string[target_type]
-                for target_type in registered_type.supported_target_types
-            ],
+            "allowedTargetTypes": allowed_target_types,
             "integrationName": integration.name if integration else None,
             "integrationId": integration.id if integration else None,
+            "inputType": input_type,
         }
 
+        if integration and registered_type.slug == "pagerduty":
+            action_response["options"] = self.fetch_pagerduty_services(organization, integration.id)
```

Comment:
@ceorourke How do we get the options for email?

Reply:
For email that appears to be handled by the [selectMembers](https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/components/selectMembers/index.tsx) component.